CRM/Pledge/Form/Pledge.php

```diff
@@ -513,28 +516,31 @@ public function postProcess() {
     }
 
     // handle Acknowledgment.
-    if (!empty($formValues['is_acknowledge']) && $pledge->id) {
+    if (!empty($formValues['is_acknowledge'])) {
 
       // calculate scheduled amount.
       $params['scheduled_amount'] = round($params['amount'] / $params['installments']);
       $params['total_pledge_amount'] = $params['amount'];
       // get some required pledge values in params.
-      $params['id'] = $pledge->id;
+      $params['id'] = $pledgeID;
       $params['acknowledge_date'] = $pledge->acknowledge_date;
       $params['is_test'] = $pledge->is_test;
       $params['currency'] = $pledge->currency;
       // retrieve 'from email id' for acknowledgement
       $params['from_email_id'] = $formValues['from_email_address'];
 
-      $this->paymentId = NULL;
       // send Acknowledgment mail.
       CRM_Pledge_BAO_Pledge::sendAcknowledgment($this, $params);
 
       $statusMsg .= ' ' . ts("An acknowledgment email has been sent to %1.<br />", [1 => $this->userEmail]);
-
+      // get the first valid payment id.
+      $nextPaymentID = PledgePayment::get()
```

Comment:
Can you please pass check_permission = FALSE here

Reply:
@monishdeb is that necessary or precautionary - the permissions for the api 'seem' OK / consistent with the form - ... I think

![image](https://user-images.githubusercontent.com/336308/182454251-540001ae-a14d-42a8-be27-79ae89895e3f.png)